### PR TITLE
defined_incidence

### PR DESCRIPTION
**Pull request by lemanska**
Hi Colm, this is the code for the study to explore prevalence and incidence of pancreatic cancer over time. 
- the code is pretty straightforward, I have taken it largely from the ADT study. Its what I have been doing to date so I am more or less confident but of course if you dont mind please have a look at the whole thing. Please dont worry about the many runs of the base code - I had a typo there and I could not figure out why it is running locally, but it is not passing the Actions test on github. 
- However, in particular, please would you be able to have a look at the way I define incidence and prevalence? would be good to know what you think and if you think I am going in the right direction. I kept the work on defining the incidence variable in this PR so that you can see my thinking. I am happy to meet you - if quicker to discuss. Big THANK YOU for agreeing to look at it.

### CODE REVIEW

File: analysis/study_definition_rates.py [hunk 1]
```diff
@@ -60,14 +60,13 @@
...
-
-    incidence=patients.with_these_clinical_events(
+    ###
+    # this is not corrrect because the desease could be diagnosed earlier, codes are entered multiple times to patinet record
+    ###
+    incid=patients.with_these_clinical_events(
         prostate_cancer_codes,
         between=[
-            "first_day_of_month(index_date)",
+            "first_day_of_month(index_date)", #diagnosed this month, will this work for measures?
```
Comment (andrewscolm) [inline analysis/study_definition_rates.py:69]: This has been used for measures [here](https://github.com/opensafely/antidepressant-prescribing-lda/blob/e50bd7479bf87c10947b0a1aec9b30b2f7518924/analysis/study_definition.py#L91)

`first_day_of_month` may be unnecessary as you are specifying the 1st of each month in your project yaml but it won't cause any harm.

File: analysis/study_definition_rates.py [hunk 2]
```diff
@@ -38,7 +38,7 @@
     ### prevalence, diagnosed any time, registered, alive and an adult in a given month
     prevalence=patients.with_these_clinical_events(
         prostate_cancer_codes,
-        on_or_before="index_date",
+        on_or_before="last_day_of_month(index_date)", #do I need the last day? for measures?
```
Comment (andrewscolm) [inline analysis/study_definition_rates.py:41]: You could make a case either way, The important thing is you are consistent about using first or last day of the month.

File: analysis/study_definition_rates.py [hunk 3]
```diff
@@ -80,6 +79,13 @@
             "incidence": 0.4
         }
     ),
+    incidence=patients.satisfying(
+        """
+        incid
+        AND incid_date = prevalence_date
+        """
+    ),## will this let me establish incidence?
```
Comment (andrewscolm) [inline analysis/study_definition_rates.py:87]: This looks correct to me. Alternatively you could se dynamic dates to check for [previous occurrences](https://github.com/opensafely/antidepressant-prescribing-lda/blob/e50bd7479bf87c10947b0a1aec9b30b2f7518924/analysis/study_definition.py#L111-L122)

**Review by andrewscolm — COMMENTED** [review]
Hi, could you talk me through the logic behind including the prevalence and incidence separately?
e.g.
```
   Measure(
        id="prevalence_rate",
        numerator="prevalence",
        denominator="population",
        group_by="population",
        small_number_suppression=True,
    ),
```
and
```
Measure(
        id="incidence_rate",
        numerator="incidence",
        denominator="population",
        group_by="population",
        small_number_suppression=True,
    ),
```
instead of 
```
Measure(
        id="incidence_rate",
        numerator="incidence",
        denominator="prevalence_rate",
        group_by="population",
        small_number_suppression=True,
    ),
```

File: analysis/study_definition_rates.py (hunk 3 above)
Comment (lemanska) [reply analysis/study_definition_rates.py:87]: Amazing, thank you, I have used this way, better than mine

**Comment by lemanska** [comment]
> 

So I am doing incidence and prevalence separately because the aim is to see the rates of new diagnosis in the population and prevalence rates in the population, these two are quite important because 1) prostate cancer is a long term condition for most - so most people do not die but 2) because we have no staging etc the two can give quite a nice story about survival. I actually added a mortality do to prostate cancer as well. I will produce a separate pull request. These three are important indicators with what happens in healthcare. e.g. t[his paper](https://pubmed.ncbi.nlm.nih.gov/34479490/) 

What you are suggesting to look for new diagnosis within people with diagnosis - i dont thik makes sense.... 
Measure(
        id="incidence_rate",
        numerator="incidence",
        denominator="prevalence_rate",
        group_by="population",
        small_number_suppression=True,
    ),
Although rates of mortality within the population with prostate cancer, i think works well?
    Measure(
        id="mortality_rate",
        numerator="died_prostate",
        denominator="prevalence",
        group_by="population",
        small_number_suppression=True,
    ),

**Pull request merged**